From 59f69162caed3781d6737d2a4bc039e2205b8470 Mon Sep 17 00:00:00 2001
From: Aleksa Sarai <cyphar@cyphar.com>
Date: Fri, 28 Oct 2022 12:44:39 +1100
Subject: [PATCH 6/7] build: only build init.lxc.static if libcap is statically
 linkable

Without setting this, the default build will fail if you don't have the
static libcap library installed (on openSUSE this is packaged separately
to libcap-devel).

Signed-off-by: Aleksa Sarai <cyphar@cyphar.com>
---
 meson.build             | 5 +++--
 src/lxc/cmd/meson.build | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 0765c9b02cd3..93572975ebe4 100644
--- a/meson.build
+++ b/meson.build
@@ -460,12 +460,13 @@ int main(int argc, char *argv[]) { return 0; };
 '''
     if libcap_static.found()
         libcap_static_linkable = cc.links(code, args: '-static', dependencies: libcap_static)
-        srcconf.set10('HAVE_STATIC_LIBCAP', libcap_static_linkable)
     else
-        srcconf.set10('HAVE_STATIC_LIBCAP', false)
+        libcap_static_linkable = false
     endif
+    srcconf.set10('HAVE_STATIC_LIBCAP', libcap_static_linkable)
 else
     libcap_static = []
+    libcap_static_linkable = false
     srcconf.set10('HAVE_LIBCAP', false)
     srcconf.set10('HAVE_STATIC_LIBCAP', false)
 endif
diff --git a/src/lxc/cmd/meson.build b/src/lxc/cmd/meson.build
index e083130254bb..238b91d0521b 100644
--- a/src/lxc/cmd/meson.build
+++ b/src/lxc/cmd/meson.build
@@ -60,7 +60,7 @@ cmd_lxc_update_config = configure_file(
     output: 'lxc-update-config')
 install_data(join_paths(project_build_root, 'src/lxc/cmd/lxc-update-config'), install_dir: bindir)
 
-if sanitize == 'none'
+if sanitize == 'none' and libcap_static_linkable
     cmd_programs += executable(
         'init.lxc.static',
         cmd_lxc_init_sources,
-- 
2.38.0

